### DISCUSSION TITLE
Real live preview

### DISCUSSION THREAD

**Discussion by fedetango** [phcode-dev/phoenix#695] [Ideas]
One cool feature that made brackets unique was the real-time HTML and CSS change. That is, the preview changed each time something was changed in the file. This to me is the ideal behavior.

**Comment by abose** [comment]
Thanks, @fedetango for the feature request.
This is the next major feature we are currently working on. Should be out before December.

**Reply by bblessatron225** [reply]
Is it possible to turn off the auto live preview?

**Reply by abose** [reply]
![image](https://github.com/phcode-dev/phoenix/assets/5336369/e18bc55d-98a9-45c8-b841-d962134b6461)
This icon will disable all the live preview highlights. What is the exact use case?

**Reply by bblessatron225** [reply]
Although the syntax highlighter is an excellent feature that I wish Microsoft Visual Studio Code had, that's not what I was talking about.

What I mean is when I open Phoenix Code, the live preview automatically starts. And I don't want that to happen. There should be a setting for those who want to enable/disable this action from happening. For example, when I'm working on my laptop, I don't want the live preview to auto-launch, because then I must close it. Granted it takes only a second more to do and that's not a big deal, but the point is I shouldn't need to have to do that. When I'm working on my desktop, sometimes I might prefer it to launch automatically, but probably not.

More suggestions:
- When I'm using my touch screen, I must tap twice to open the next menu in the ribbon. When I'm using my mouse and I leave the ribbon, the menu stays open until I click outside of it.
- Where are the settings or preferences? And why are the preferences unchangeable?

I have more suggestions, but for now, these are the most related.

**Reply by abose** [reply]
1. For live preview, would a setting to open live preview on project start do? Most of our users want live preview to always come up, that is why its optimized for that by default.
2. Preference currently is just a json file that can be found in file menu - efit preference file. We will be adding a more streamlined preferences settings panel in the coming releases.
3. The touch screen bug needs investigation. Thanks for reporting.

Please do drop additional feature requests, usually critical feature requests or fixes directly from users get addressed in the following release.

**Comment by abose** [comment]
The live preview is now available in Phoenix.

**Reply by Ru571k** [reply]
This is really good. But is it possible to add or convert live preview to visual editor like tinymce

**Reply by abose** [reply]
Eventually, it should be possible. But working hard to get native builds for windows Mac and Linux rn. Can you create a feature request in issues and note down your desired features and use cases. It will help prioritize it.

**Comment by fedetango** [comment]
Good news! Thank you 😃

**Comment by paulmz** [comment]
I thought the Live Preview was updated to be like Brackets? I just installed Version 4.0.3 (20250131.060146) and I'm not seeing instant changes in HTML or CSS in the preview window.

This and the annoyance that Phoenix uses a single instance of Chrome for live preview honestly makes Phoenix unusable for me. If I already have a Chrome window open, Brackets opens Live Preview in a totally separate Chrome instance making it possible for me to still use Chrome separately from the preview. I'll be sticking with Brackets until Live Preview works EXACTLY as it does in Brackets.

**Reply by devvaannsh** [reply]
@paulmz, Phoenix Live Preview is the same as Brackets. If you're not seeing instant changes in HTML or CSS, it might be due to something else. Try this: open a small HTML file, then open Live Preview. At the top of the Live Preview panel, there's a refresh button click on it. After that, you should start seeing instant changes.

About the Chrome instance part, I didn't fully get what you meant. Could you share a recording showing the difference between how it works in Brackets and Phoenix? As far as I know, Live Preview in both is exactly the same, with Phoenix just having a few extra features that shouldn't affect this.
